plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixMetadata.java

```diff
@@ -86,6 +91,22 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
                 .orElse(null);
     }
 
+    @Override
+    public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
+    {
+        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
+        List<LocalProperty<ColumnHandle>> sortingProperties = tableHandle.getSortOrder()
+                .map(properties -> properties
+                        .stream()
+                        .map(item -> (LocalProperty<ColumnHandle>) new SortingProperty<ColumnHandle>(
```

Comment:
the cast could be avoided by using return type `List<? extends LocalProperty<ColumnHandle>>`

Reply:
But then you can't pass sortingProperties to ConnectorTableProperties.<init>

Reply:
i see, the cast is fine then

Reply:
It did catch me by surprise first that the version with the if-statement did not need this cast to compile, since the final assignment is the same - probably an artifact of the details of how the lambda expressions work in Java.